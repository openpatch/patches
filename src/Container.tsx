/** @jsx jsx */
import { ReactNode } from "react";
import { jsx, css } from "@emotion/react";
import { maxWidth, MaxWidthProps } from "styled-system";
import styled from "@emotion/styled";

export type ContainerProps = MaxWidthProps;

export const Container = styled.div<ContainerProps>(
  {
    margin: "0 auto",
  },
  maxWidth
);

Container.defaultProps = {
  maxWidth: "medium",
};
