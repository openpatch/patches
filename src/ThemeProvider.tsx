import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
