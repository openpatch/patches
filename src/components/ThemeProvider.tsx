import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
