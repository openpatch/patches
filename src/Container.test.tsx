import { render } from "@testing-library/react";
import { Container, ContainerProps } from "./Container";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: ContainerProps) =>
  render(
    <ThemeProvider>
      <Container {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
