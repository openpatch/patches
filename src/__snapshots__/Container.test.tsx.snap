// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  margin: 0 auto;
  max-width: 940px;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;
